primitives/nomad/nomad-core/src/lib.rs

```diff
@@ -1,5 +1,33 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
+/// Variadic macro used by `keccak256_concat` internally.
+#[macro_export]
+macro_rules! keccak256_concat_update {
+	($hasher:ident, $e:expr) => {{
+		$hasher.update($e.as_ref());
+	}};
+
+	($hasher:ident, $e:expr, $($es:expr),+) => {{
+		$hasher.update($e.as_ref());
+		$crate::keccak256_concat_update!($hasher, $($es),+);
+	}};
+}
+
+/// Calculates the Kecck 256 of arguments with NO extra allocations to join inputs.
+#[macro_export]
+macro_rules! keccak256_concat{
+	($($arg:tt)*) => {{
+		{
+			use tiny_keccak::Hasher as _;
+			let mut output = [0u8; 32];
+			let mut hasher = tiny_keccak::Keccak::v256();
+			$crate::keccak256_concat_update!(hasher, $($arg)*);
```

Comment:
I have no experience with macros, but wouldn't `Vec` argument instead of vararg and for loop allow using this as fn instead of macro? If thats desired anyway :D

Reply:
This macro has two main objectives:
- Reduce code duplication (which can also be achieve using `vec` as you suggested).
- Remove the extra memory allocation and/or extra copies. Using `vec` will require an extra heap allocation (required by `vec` itself), and deal with references and lifetimes to avoid copies between source objects into `vec`.